pandas/sparse/array.py

```diff
@@ -420,7 +459,12 @@ def astype(self, dtype=None):
         dtype = np.dtype(dtype)
         if dtype is not None and dtype not in (np.float_, float):
             raise TypeError('Can only support floating point data for now')
-        return self.copy()
+
+        if self.dtype == dtype:
+            return self.copy()
+        else:
+            return self._simple_new(self.sp_values.astype(dtype),
+                                    self.sp_index, float(self.fill_value))
```

Comment:
Yeah, it should be covered in #13849 (though i'm thinkinn to split small PRs though...). 

Because currently ``astype`` only supports to coerce ``float``, it is a temp workaround not to break others.